pagopa/pn-bff
# prevent index out of bound error - pn-9829

## Short description

Fixed out of bound error when moving the items from the Delivered status to the Delivering one. This error happens when the status has more elements in relatedTimelineElements array than in timeline array.

## List of changes proposed in this pull request

- Added check into the populateMacroSteps method

## How to test

Call the received notification detail api with user colombo and iun ZMLJ-LELZ-PJQW-202405-D-1 -> check that the api doesn't return error
